### PR TITLE
adding cloudsqlproxy wrapper and use exitdir to signal when the pod should terminate

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
- adding cloudsqlproxy wrapper and use exitdir to signal when the pod should terminate

in our public instance we use cloudsql and the cloudsql-proxy to connect in the DB
however in the `createdb` job the cloudsql run forever and never finish because it is not aware of the main container that creates the db, using the `https://github.com/chainguard-dev/exitdir` we can signal from `createdb` to cloudproxy when we are ready and then the job the gracefully finish

This is the first part of the fix and the second is to update https://github.com/sigstore/helm-charts/blob/main/charts/trillian/templates/createdb/createdb-job.yaml to use this image

This will fix: https://github.com/sigstore/public-good-instance/issues/965

### CODE REVIEW

**Comment by cpanato** [comment]
@vaikas @priyawadhwa let me know if this makes sense as initial iteration

**Review by vaikas — DISMISSED** [review]
Super! Thanks for doing this!

**Review by bobcallaway — APPROVED** [review]

**Review by k4leung4 — APPROVED** [review]
sweet

File: .ko.yaml [hunk 1]
```diff
@@ -1,182 +1,200 @@
+---
 defaultBaseImage: cgr.dev/chainguard/alpine-base:latest
+baseImageOverrides:
+  github.com/sigstore/scaffolding/cmd/cloudsqlproxy: gcr.io/cloudsql-docker/gce-proxy:1.32.0-alpine
```
Comment (k4leung4) [inline .ko.yaml:4]: should we use 1.33.1 here, as that is the default in the helm chart

https://github.com/sigstore/helm-charts/blob/fdca671033e29dc58e15234644cf511d146f72db/charts/trillian/Chart.yaml#LL37C14-L37C118

Comment (cpanato) [reply .ko.yaml:4]: yes, thanks for the catch!

Comment (cpanato) [reply .ko.yaml]: done, PTAL

**Review by vaikas — APPROVED** [review]

**Comment by cpanato** [comment]
will continue the work

**Pull request merged**